From 785075f812801108402ec2363bdf0ccaec6402e7 Mon Sep 17 00:00:00 2001
From: "Konstantin Kirik (snegovick)" <snegovick@uprojects.org>
Date: Mon, 5 Jul 2021 23:44:01 +0300
Subject: [PATCH 083/106] BMLK-39: Implement ssh-based repo operations

Add before-script which reconfigures default
gitlab git flow with ssh-based flow, which allows
to make large repo clones more reliable
---
 .gitlab-ci.yml | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index 105346dfb1a8..e1449b543cd8 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -1,6 +1,5 @@
 variables:
-  GIT_DEPTH: 200
-  GIT_STRATEGY: clone
+  GIT_STRATEGY: none
 
 .build-kernel:
   stage: build
@@ -10,6 +9,24 @@ variables:
     BOARD: bm
   tags:
     - x86_64
+  before_script:
+    - if [ -d "${CI_PROJECT_DIR}" ]; then rm -rf "${CI_PROJECT_DIR}"; fi  # step 1: recreate project directory
+    - mkdir -p "${CI_PROJECT_DIR}"
+    - cd "${CI_PROJECT_DIR}"
+    - mkdir -p ~/.ssh # step 2 : set up ssh environment
+    - cat $deploy_id | base64 --decode > ~/.ssh/id_rsa
+    - echo "$deploy_id"
+    - ls -lah $deploy_id
+    - chmod 600 ~/.ssh/ -R
+    - eval "$(ssh-agent -s)"
+    - md5sum ~/.ssh/id_rsa
+    - ssh-add ~/.ssh/id_rsa
+    - ssh-keyscan -H 'gitlab.t-platforms.ru' >> ~/.ssh/known_hosts
+    - git init # step 3 : fetch
+    - git remote add origin ssh://git@${CI_SERVER_HOST}/${CI_PROJECT_PATH}.git
+    - git fetch origin "${CI_COMMIT_SHA}" --depth 200
+    - git reset --hard FETCH_HEAD
+    - git fetch --tags
   script:
     - ARCH=arm64 make $DEFCONFIG
     - SKIP=$(git describe --tags --no-abbrev | wc -m)
-- 
2.33.0

